bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java

```diff
@@ -277,34 +265,32 @@ public MongoCryptContext createExplicitEncryptionContext(final BsonDocument docu
             throwExceptionFromStatus();
         }
 
-        boolean success;
-
         if (options.getKeyId() != null) {
             try (BinaryHolder keyIdBinaryHolder = toBinary(ByteBuffer.wrap(options.getKeyId().getData()))) {
-                success = mongocrypt_ctx_setopt_key_id(context, keyIdBinaryHolder.getBinary());
-                if (!success) {
-                    MongoCryptContextImpl.throwExceptionFromStatus(context);
-                }
+                configure(() -> mongocrypt_ctx_setopt_key_id(context, keyIdBinaryHolder.getBinary()), context);
             }
         } else if (options.getKeyAltName() != null) {
             try (BinaryHolder keyAltNameBinaryHolder = toBinary(new BsonDocument("keyAltName", new BsonString(options.getKeyAltName())))) {
-                success = mongocrypt_ctx_setopt_key_alt_name(context, keyAltNameBinaryHolder.getBinary());
-                if (!success) {
-                    MongoCryptContextImpl.throwExceptionFromStatus(context);
-                }
+                configure(() -> mongocrypt_ctx_setopt_key_alt_name(context, keyAltNameBinaryHolder.getBinary()), context);
             }
         }
 
-        success = mongocrypt_ctx_setopt_algorithm(context, new cstring(options.getAlgorithm()), -1);
-        if (!success) {
-            MongoCryptContextImpl.throwExceptionFromStatus(context);
+        if (INDEXED.equalsIgnoreCase(options.getAlgorithm())) {
```

Comment:
If no algorithm is set, does `getAlgorithm()` return null? Is that OK in this comparison?